gazebo_ros/test/test_gazebo_ros_init.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <unistd.h>
+#include <stdlib.h>
+
+#include <gtest/gtest.h>
+#include <rclcpp/rclcpp.hpp>
+#include <std_msgs/msg/string.hpp>
+
+#include <chrono>
+#include <memory>
+#include <thread>
+
+using namespace std::chrono_literals;
+
+class GazeboRosInitTest : public ::testing::Test
+{
+public:
+  void TearDown();
+  int pid_{-1};
+};
+
+void GazeboRosInitTest::TearDown()
+{
+  // If fork failed, don't need to do anything
+  if (pid_ < 0) {
```

Comment:
Wonder if this test could be refactored to avoid the duplicate code for forking / subscribing / verifying messages received that is in the other tests.

Reply:
Yeah you're right I'll do that. I had started writing a completely different test but then they ended up very similar so it should be possible to move this to `test_plugins`.

Reply:
Done on https://github.com/ros-simulation/gazebo_ros_pkgs/pull/776/commits/10be195185bca8015c02a391f628cfbabf944216